core/shoplist_engine/aggregator.py

```
@@ -0,0 +1,123 @@
+
+RU: Этот модуль агрегирует IngredientSpec по food_id, суммируя количества
+в базовых единицах (G, ML, PCS). Входные данные должны быть уже нормализованы
+к базовым единицам.
+
+All functions are pure (no I/O, env, time, random dependencies).
+RU: Все функции чистые (нет зависимостей от I/O, env, time, random).
+"""
+
+from __future__ import annotations
+
+from collections import defaultdict
```

Flake8 (7.3.0)

[error] 'collections.defaultdict' imported but unused

(F401)